### PR TITLE
Error when creating a report from the Facility Search Page

### LINKED ISSUE #192: Error Exporting Reports from Facility Search Page
Application freezes when user wants to export results to a report from "Facility Search" Page. 
Application is supposed to go to the Reports page to show a preview of the report. This Fails.

### PR DESCRIPTION
**Pull request by tom-karasch**
Originally when clicking the **Create Report** button on the **Facility Search** page sent you to the **Report** page giving the user a preview of the report to be sent to Excel, along with the search criteria. However, in production the app failed to go to the **Report** page and brought up a login screen instead, which did not work. 
So, to correct the problem, temporarily, instead of sending the user to the report page, the user now sees the button, renamed **Export to Excel** and directly generates the report file ( a **.csv** file of the requested Facility Detailed List).

### CODE REVIEW

File: FMS/Pages/Facilities/Index.cshtml.cs [hunk 1]
```diff
@@ -1,75 +1,112 @@
...
+    {
+        private readonly IFacilityRepository _repository;
+        private readonly ISelectListHelper _listHelper;
+
+        // "Spec" is the Facility DTO bound to the HTML Page elements
+        public FacilitySpec Spec { get; set; }
+
+        // "ReportSpec" is the Facility DTO Bound to Export Results to Excel (csv file)
+        [BindProperty]
+        public FacilitySpec ReportSpec { get; set; }
```
Comment (dougwaldron) [inline FMS/Pages/Facilities/Index.cshtml.cs]: Don't create a new `FacilitySpec` property here. Bind the existing `Spec` instead.

File: FMS/Pages/Facilities/Index.cshtml.cs [hunk 2]
```diff
@@ -1,75 +1,112 @@
...
+
+        // List of facilities resulting from the search
+        public IPaginatedResult FacilityList { get; private set; }
+
+        // Detailed Facility List to go to a report
+        public IReadOnlyList<FacilityDetailDto> FacilityReportList { get; private set; }
```
Comment (dougwaldron) [inline FMS/Pages/Facilities/Index.cshtml.cs]: Since this property is only used in the `OnPostAsync` method, move it there as a local variable instead.

**Review by dougwaldron — CHANGES_REQUESTED** [review]
@tom-karasch Good start on this, and I agree this is the best way to address this issue. Besides avoiding the original authorization bug, it will remove a step for the user along with the server overhead of pulling the same data twice on the Reports page. 

Please address the following issues with the current PR:

1. The exported spreadsheet does not match the search results. You're creating a new, empty `FacilitySpec` for the report instead of using the existing `Spec` property that represents the user's  search criteria. Did you look at this?

2. Delete all commented out code. Commented code doesn't add value and distorts the version control code diffs.

3. Delete the "Report" page; it's no longer used.

4. In the body of your PR comment, add the following line. This links the PR with the original issue so that merging the PR will automatically close it.

    `fixes #192`

By the way, you don't need to create a new branch or PR or anything when you implement these fixes. Any additional commits you push to this branch will automatically update this PR.

**Comment by tom-karasch** [comment]
Corrections to initial Reporting fix.
Removed "Report" Page and folder from Project.
Fixed use of Public property "Spec" and removed commented code.
Changed "Export" Button and changed "Get" to "Post" in HTML.

fixes #192

**Review by dougwaldron — APPROVED** [review]
Thanks! I can confirm that the export matches the search results. This is ready for QA.

A couple minor cleanup items: Not sure how it happened, but instead of deleting the Report page files, you just hid them from the VS project. I'll fix that shortly.

I also noticed that we no longer need to disable search form inputs when clicking the export button. I've fixed that as well.

**Comment by tom-karasch** [comment]
Thanks, Doug.
I did not realize that the search form inputs were disabled after Export. 
I assumed removing the Report page from the project would mean that they are no longer published. I would like to revisit the export preview Page when adding direct to ".xls" format. So, I didn't see the harm in removing them from the project without deleting them. 
So, now I will Merge the pull request.
Should I then publish to both Dev and UAT or just UAT?

**Comment by dougwaldron** [comment]
Oh, I see. You're right that the report page wouldn't publish once it's removed from the project. But still, leaving them in is just like leaving in commented code. If you ever want to restore those files, just revert commit 66e272a7755adc879f7a7647026f284c85cf6c7d.

Go ahead and publish to dev and send to Tamika for QA. Thanks.

**Pull request merged**